Quotient/events/roommessageevent.cpp

```diff
@@ -219,6 +219,49 @@ QString RoomMessageEvent::replacedBy() const
     // clang-format on
 }
 
+bool RoomMessageEvent::isReply() const
+{
+    const auto relations = contentPart<QJsonObject>("m.relates_to"_ls);
+    if (!relations.isEmpty()) {
+        const bool hasReplyRelation = relations.contains("m.in_reply_to"_ls);
+        bool isFallingBack = relations["is_falling_back"_ls].toBool();
+        return hasReplyRelation && !isFallingBack;
+    }
+    return false;
+}
+
+bool RoomMessageEvent::isReplyIncludingFallbacks() const
+{
+    return contentPart<QJsonObject>("m.relates_to"_ls).contains("m.in_reply_to"_ls);
+}
+
+QString RoomMessageEvent::replyEventId() const
+{
+    return contentJson()["m.relates_to"_ls].toObject()["m.in_reply_to"_ls].toObject()["event_id"_ls].toString();
+}
```

Comment:
I would go with something more generic here. I don't quite like "parent" and "child" event naming the spec uses, but I can accept it as well if my suggestion doesn't ring true.
```suggestion
QString RoomMessageEvent::upstreamEventId() const
{
    const auto relation = relatesTo();
    return relation ? relation.eventId : QString();
}
```